src/server/lib/okta/tokens.ts

```diff
@@ -46,17 +46,47 @@ const issuerDomain = () => {
 };
 
 /**
- * @name oauthTokenVerifier
+ * @name oauthAccessTokenVerifier
  * @type {OktaJwtVerifier}
- * @description Okta JWT Verifier
+ * @description Okta JWT Verifier - Access Token
+ *
+ * This verifier is used to verify the access token, as Okta JWT Verifier doesn't verify
+ * the clientId claim access, we need to assert it manually for the access token (`cid` claim) which is different
+ * from the ID token (`aud` claim) where Okta verifies it through the `expectedClientId` parameter in `verifyIdToken`.
+ *
+ * This is because in Gateway we want to make sure the access token is for this application, and not for any other before
+ * we send it to other APIs.
+ */
+const oauthAccessTokenVerifier: Pick<OktaJwtVerifier, 'verifyAccessToken'> =
+	new OktaJwtVerifier({
+		issuer: joinUrl(issuerDomain(), '/oauth2/', okta.authServerId),
+		clientId: okta.clientId,
+		assertClaims: {
+			cid: okta.clientId,
+		},
+	});
```

Comment:
To avoid repetition in both verifiers, could this be implemented by a common method that then does the `assertClaim` in `oauthAccessTokenVerifier` only?

Reply:
I tried having a look, but this doesn't seem possible to have a common method for both.

However to avoid config repetion, I have separate the `issuer` and `clientId` into its own shared options object, and use that as the base for the verifier instantiation